lib/id3c/cli/command/etl/manifest.py

```diff
@@ -95,55 +95,71 @@ def etl_manifest(*, db: DatabaseSession):
+            sample_identifier = find_identifier(db, sample_barcode) if sample_barcode else None
             collection_barcode = manifest_record.document.pop("collection", None)
             collection_identifier = find_identifier(db, collection_barcode) if collection_barcode else None
 
+            # Skip a record if it has no sample_identifier and no collection_identifier
+            if not sample_identifier and not collection_identifier:
+                LOG.warning(f"Skipping record «{manifest_record.id}» because it has no known sample barcode or collection barcode")
+                mark_skipped(db, manifest_record.id)
+                continue
+
+            # Skip a record if it has a sample barcode but the barcode doesn't match an identifer
+            if sample_barcode and not sample_identifier:
+                LOG.warning(f"Skipping sample with unknown sample barcode «{sample_barcode}»")
+                mark_skipped(db, manifest_record.id)
+                continue
+
+            # Skip a record if it has a collection barcode but the barcode doesn't match an identifer
             if collection_barcode and not collection_identifier:
                 LOG.warning(f"Skipping sample with unknown collection barcode «{collection_barcode}»")
                 mark_skipped(db, manifest_record.id)
                 continue
 
-            assert not collection_identifier \
-                or collection_identifier.set_name in expected_identifier_sets["collections"], \
-                    f"Collection identifier found in set «{collection_identifier.set_name}», not {expected_identifier_sets['collections']}" # type: ignore
+            # Validate the sample identifer and assert if a record fails
+            if sample_identifier:
+                if (manifest_record.document.get("sample_type") and
+                    manifest_record.document["sample_type"] == "rdt"):
+                    assert sample_identifier.set_name in expected_identifier_sets["rdt"], \
+                        (f"Sample identifier found in set «{sample_identifier.set_name}»," +
+                        f"not {expected_identifier_sets['rdt']}")
+                else:
+                    assert sample_identifier.set_name in expected_identifier_sets["samples"], \
+                        (f"Sample identifier found in set «{sample_identifier.set_name}», " +
+                        f"not {expected_identifier_sets['samples']}")
+
+            # Skip a record if the collection identifier is from an unexpected set
+            if collection_identifier and collection_identifier.set_name not in expected_identifier_sets["collections"]:
+                LOG.warning(f"Skipping sample because collection identifier found in set «{collection_identifier.set_name}», not \
+                    {expected_identifier_sets['collections']}")
+                mark_skipped(db, manifest_record.id)
+                continue
```

Comment:
I feel like this barcode skipping block fits better with the code where we arre "Skipping sample with unknown collection barcode «{collection_barcode}»." If it makes sense, could you please move it up there?

Reply:
I put the collection check above the sample check.